src/SunburstVisual.ts

```diff
@@ -0,0 +1,443 @@
+                : 0;
+
+            const newSunNode: SunburstSlice = {
+                name: originParentNode.name,
+                value: Math.max(valueToSet, 0),
+                selector: selectionId,
+                key: selectionId
+                    ? selectionId.getKey()
+                    : null,
+                total: valueToSet,
+                children: []
+            };
+
+            if (originParentNode.value) {
+                newSunNode.color = color
+                    ? color
+                    : colors.getColor(originParentNode.value.toString()).value;
+            }
+
+            data.total += newSunNode.value;
+            newSunNode.children = [];
+
+            if (originParentNode.children && originParentNode.children.length > 0) {
+                newSunNode.tooltipInfo = this.getTooltipData(<string>originParentNode.value, -1);
+                for (const child of originParentNode.children) {
+                    const newChild: SunburstSlice = this.covertTreeNodeToSunBurstNode(
+                        child,
+                        newSunNode,
+                        colors,
+                        pathIdentity,
+                        data,
+                        newSunNode.color,
+                        visualHost,
+                        level + 1);
+
+                    newSunNode.children.push(newChild);
+                    newSunNode.total += newChild.total;
+                }
+            }
+            else {
+                newSunNode.tooltipInfo = this.getTooltipData(
+                    <string>originParentNode.value,
+                    valueToSet);
+            }
+
+            if (sunburstParentNode) {
+                newSunNode.parent = sunburstParentNode;
+            }
+
+            return newSunNode;
+        }
+
+        private getTooltipData(displayName: string, value: number): VisualTooltipDataItem[] {
+            return [{
+                displayName,
+                value: value < 0
+                    ? ""
+                    : value.toString()
+            }];
+        }
+
+        private parseSettings(dataView: DataView): SunburstSettings {
+            return SunburstSettings.parse<SunburstSettings>(dataView);
+        }
+
+        // Get all parents of the node
+        private static getTreePath(node: SunburstSlice): SunburstSlice[] {
+            const path: SunburstSlice[] = [];
+            let current: SunburstSlice = node;
+
+            while (current.parent) {
+                path.unshift(current);
+                current = current.parent;
+            }
+
+            return path;
+        }
+
+        private calculateLabelPosition(): void {
+            const innerRadius: number = Math.min.apply(
+                null,
+                this.data.root.children.map((x: SunburstSlice) => this.arc.innerRadius()(x, undefined))
+            );
+            const commonLineHeight: number = 1 + 1 * Sunburst.PercentageFontSizeMultiplier +
```

Comment:
```1 * Sunburst.PercentageFontSizeMultiplier``` doesn't make sense. Please use ```Sunburst.PercentageFontSizeMultiplier``` instead. Please include a comment that describes the hidden logic.